test/routes.js

```diff
@@ -1702,6 +1749,9 @@ test('promise rejection from onDisconnect is logged', t => {
   const error = new Error('error')
 
   const app = Fastify()
+
+  // override `app.log` to avoid polluting other tests
+  app.log = Object.create(app.log)
```

Comment:
I don't think this is needed

Reply:
I believe it is. when we do not provide `logger` instance (or config) to `Fastify` factory, it use [abstract-logging](https://www.npmjs.com/package/abstract-logging) library. The important part is that it is the same logger object instance for every fastify instance.

When this test override `app.log.error` it do it for all other future tests that use `Fastify()` factory. It was not a problem since now because of tests order and behavior, but it started to be with current code.

Reply:
Ah awesome ok!